### PR TITLE
Shiftleft

### PR DESCRIPTION
**Pull request by mvr320**
I verified the reported issues and none of those seem to be completely false issues.

### CODE REVIEW

File: lib/lib.error.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ logmsg ()
 
 	msglevel="$1"; shift
 	msec=`date '+%N'`
-	stamp="[`date '+%b %d %T'`.`printf "%.3s" $msec`] $PROGNAME[$$]:"
+	stamp="[`date '+%b %d %T'`.`printf "%.3s" $msec`] ${PROGNAME}[$$]:"
```
Comment (mvr320) [inline lib/lib.error.sh:23]: @eldering I think this is more correct but I'm not 100% sure if this has no unwanted side effects?

**Comment by mvr320** [comment]
The last error:
```
Argument mixes string and array. Use * or separate argument.

if [ "$@" ]; then
		logmsg $LOG_ERR "error: $@"
```

in lib/lib.error.sh is I think already discussed in code but I'm not sure if it can currently be handled or if general support for `echo -e` is still in a bad state?

File: lib/lib.error.sh (hunk 1 above)
Comment (eldering) [reply lib/lib.error.sh:23]: No, that's fine. Actually less likely to have unwanted side effects, since it's clearer what the variable name is.

**Comment by eldering** [comment]
> The last error:
> 
> ```
> Argument mixes string and array. Use * or separate argument.
> 
> if [ "$@" ]; then
> 		logmsg $LOG_ERR "error: $@"
> ```
> 
> in lib/lib.error.sh is I think already discussed in code but I'm not sure if it can currently be handled or if general support for `echo -e` is still in a bad state?

I think both of these should be `$*`, there's a subtle difference (that I also used to do wrong), see `man bash`.

**Review by nickygerritsen — APPROVED** [review]

**Pull request merged**